pkgs/applications/editors/emacs/build-support/melpa.nix

```diff
@@ -62,17 +64,17 @@ in
     recipe: Optional MELPA recipe.
     Default: a minimally functional recipe
   */
-, recipe ? (writeText "${pname}-recipe" ''
-    (${ename} :fetcher git :url ""
-              ${lib.optionalString (files != null) ":files ${files}"})
+, recipe ? (writeText "${finalAttrs.pname}-recipe" ''
+    (${finalAttrs.ename} :fetcher git :url ""
+              ${lib.optionalString (finalAttrs.files or null != null) ":files ${finalAttrs.files}"})
   '')
 , preUnpack ? ""
 , postUnpack ? ""
 , meta ? {}
 , ...
 }@args:
 
-genericBuild ({
+{
 
   elpa2nix = ./elpa2nix.el;
   melpa2nix = ./melpa2nix.el;
```

Comment:
Should these two respect the value passed to `melpaBuild`?  I decide to ignore those values because I consider them as implementation detail of this build helper.

```nix
melpaBuild {
  elpa2nix = ./some-custom-file.el
  ...
}
```

The same question applies to `packageBuild`, `melpaVersion`

Any thoughts? @AndersonTorres 

related: https://github.com/NixOS/nixpkgs/pull/330589/files#r1697791893